mods/cnc/maps/gdi01/gdi01.lua

```diff
@@ -49,11 +49,10 @@ Reinforce = function(units)
 end
 
 CheckForBase = function()
-	baseBuildings = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
-		return actor.Type == "fact" or actor.Type == "pyle" or actor.Type == "nuke"
-	end)
-
-	return #baseBuildings >= 3
+	local barracks = player.GetActorsByType("pyle")
+	if #barracks > 0 then
+		return true
```

Comment:
I find the current code easier to read - I can change it, but is it worth it in terms of runtime efficiency? (that function is only called once per second, and the GetActorsByType call is probably the bottleneck there)